ToolBox/CoarseUniverseGenerator/CoarseUniverseGeneratorProgram.cs

```diff
@@ -13,375 +13,332 @@
+            // The dates right after the acquisition, valid fine fundamental data for AABA are still under the former ticker folder.
+            // Therefore if no fine fundamental data is found in the 'fundamental/fine/aaba' folder, it searches into the 'yhoo' folder.
+            if (mapFile != null && mapFile.Count() > 2 && !hasFundamentalDataForDate)
             {
-                var excludedSymbols = File.ReadLines(exclusionsFile).Select(x => x.Trim()).Where(x => !x.StartsWith("#"));
-                exclusions = new HashSet<string>(excludedSymbols, StringComparer.InvariantCultureIgnoreCase);
-                Log.Trace("CoarseGenerator.ReadExclusionsFile(): Loaded {0} symbols into the exclusion set", exclusions.Count);
+                var previousTicker = mapFile.LastOrDefault(m => m.Date < date)?.MappedSymbol;
+                if (previousTicker != null)
+                {
+                    var previousTickerFineFundamentalFolder = Path.Combine(fineFundamentalFolder.FullName, previousTicker);
+                    if (Directory.Exists(previousTickerFineFundamentalFolder))
+                    {
+                        var previousTickerFineAvailableDates = Directory.GetFiles(previousTickerFineFundamentalFolder, "*.zip")
+                            .Select(f => DateTime.ParseExact(Path.GetFileNameWithoutExtension(f), DateFormat.EightCharacter, CultureInfo.InvariantCulture))
+                            .ToList();
+                        hasFundamentalDataForDate = previousTickerFineAvailableDates.Where(d => d >= firstDate).Any(d => date.AddMonths(-1) <= d && d <= date);
+                    }
+                }
             }
-            return exclusions;
+
+            return hasFundamentalDataForDate;
         }
 
         /// <summary>
-        /// Resolves the start date that should be used in the <see cref="ProcessDailyFolder"/>. This will
-        /// be equal to the latest file date (20150101.csv) plus one day
+        /// Parses the daily file.
         /// </summary>
-        /// <param name="coarseDirectory">The directory containing the coarse files</param>
-        /// <returns>The last coarse file date plus one day if exists, else DateTime.MinValue</returns>
-        public static DateTime GetLastProcessedDate(string coarseDirectory)
+        /// <param name="dailyFile">The daily file.</param>
+        /// <returns></returns>
+        private static List<TradeBar> ParseDailyFile(FileInfo dailyFile)
         {
-            var lastProcessedDate = (
-                from coarseFile in Directory.EnumerateFiles(coarseDirectory)
-                let date = TryParseCoarseFileDate(coarseFile)
-                where date != null
-                // we'll start on the following day
-                select date.Value.AddDays(1)
-                ).DefaultIfEmpty(DateTime.MinValue).Max();
-
-            return lastProcessedDate;
-        }
+            var scaleFactor = 1 / 10000m;
```

Comment:
I'm thinking it would be nice if we could use the `TradeBar.ParseEquity()` directly so that we don't have to handle (and know) about this scale factor which could change

Reply:
I see some disadvantages with the approach. Mainly, the overhead of create the `SubscriptionDataConfig` needed in order to use that method.